lib/mail/matchers/has_sent_mail.rb

```diff
@@ -51,6 +51,10 @@ def bcc(recipient_or_list)
         self
       end
 
+      def with_attachments(*attachments)
+        @attachments = attachments
+        self
+      end
```

Comment:
passing `attachment_or_list` instead of `*attachments` disallows us to pass no argument to this matcher.  I like the splat better, what are your thoughts?